utils_cv/detection/model.py

```diff
@@ -285,63 +380,138 @@ def plot_precision_loss_curves(
         """ Plot training loss from calling `fit` and average precision on the
         test set. """
         fig = plt.figure(figsize=figsize)
-        ax1 = fig.add_subplot(111)
+        ap = {k: [dic[k] for dic in self.ap] for k in self.ap[0]}
 
-        ax1.set_xlim([0, self.epochs - 1])
-        ax1.set_xticks(range(0, self.epochs))
-        ax1.set_title("Loss and Average Precision over epochs")
-        ax1.set_xlabel("epochs")
-        ax1.set_ylabel("loss", color="g")
-        ax1.plot(self.losses, "g-")
+        for i, (k, v) in enumerate(ap.items()):
 
-        ax2 = ax1.twinx()
-        ax2.set_ylabel("average precision", color="b")
-        ax2.plot(self.ap, "b-")
+            ax1 = fig.add_subplot(1, len(ap), i+1)
 
-    def evaluate(self, dl: DataLoader = None) -> CocoEvaluator:
+            ax1.set_xlim([0, self.epochs - 1])
+            ax1.set_xticks(range(0, self.epochs))
+            ax1.set_title("Loss and Average Precision over epochs")
+            ax1.set_xlabel("epochs")
+            ax1.set_ylabel("loss", color="g")
+            ax1.plot(self.losses, "g-")
+
+            ax2 = ax1.twinx()
+            ax2.set_ylabel("average precision for ".format(k), color="b")
+            ax2.plot(v, "b-")
+
+    def evaluate(self, dl: DataLoader = None) -> Union[CocoEvaluator, None]:
         """ eval code on validation/test set and saves the evaluation results in self.results. """
         if dl is None:
+            if not self.dataset:
+                return
             dl = self.dataset.test_dl
         self.results = evaluate(self.model, dl, device=self.device)
         return self.results
 
+    def _transform(
+        self,
+        im: Union[str, Path, Image.Image, np.ndarray],
+    ) -> torch.Tensor:
+        """ Convert the image to the format required by the model. """
+        transform = transforms.Compose([transforms.ToTensor()])
+        im = transform(im)
+        if self.device:
+            im = im.to(self.device)
+        return im
+
+    @classmethod
+    def _apply_threshold(
```

Comment:
there are now two `_apply_threshold` methods. Can you delete the old one?

Also, out of curiousity, why make this a cls method? We're never going to call `DetectionLearner._apply_threshold` are we?

Reply:
I have no idea.  My thinking is since no other places need to use this method and it is only used in `DetectionLearner`, making it a method of `DetectionLearner` would be easier to organize. That is also why a underscore is prefixed.  What do you think?

Reply:
I agree with JS. This function does not dependent on anything in the function, plus it makes it harder to review since I cannot see what the diff is to before. Should probably remove from here again and place where it used to be. @jiata?

Reply:
Reverted and added mask related code and testing.